tools/show-stylus.py

```diff
@@ -89,36 +82,75 @@ def main():
 
         print("Please put tool in proximity")
 
-        while True:
-            for event in d.events():
-                if event.matches(libevdev.EV_ABS.ABS_MISC):
-                    if event.value != 0:
-                        current_type = event.value
-                        dirty = True
-                elif event.matches(libevdev.EV_MSC.MSC_SERIAL):
-                    if event.value != 0:
-                        current_serial = event.value & 0xFFFFFFFF
+        try:
+            while True:
+                for event in d.events():
+                    if event.matches(libevdev.EV_ABS.ABS_MISC):
+                        if event.value != 0:
+                            current_type = event.value
+                            dirty = True
+                    elif event.matches(libevdev.EV_MSC.MSC_SERIAL):
+                        if event.value != 0:
+                            current_serial = event.value & 0xFFFFFFFF
+                            dirty = True
+                    elif event.code in tool_bits:
+                        # print(f'Current prox: {event.value}')
+                        in_prox = event.value != 0
                         dirty = True
-                elif event.code in tool_bits:
-                    # print(f'Current prox: {event.value}')
-                    in_prox = event.value != 0
-                    dirty = True
-                elif event.matches(libevdev.EV_SYN.SYN_REPORT) and dirty:
-                    dirty = False
-                    # len(styli) lines up, 1000 to the left and clear line
-                    print(f"\x1B[{len(styli)}A\x1B[1000D\x1B[K", end="")
-                    styli[(current_type, current_serial)] = in_prox
-                    for s, prox in styli.items():
-                        tid, serial = s
-                        print(
-                            f"Tool id {tid:#x} serial {serial:#x} in-proximity: {prox} "
-                        )
+                    elif event.matches(libevdev.EV_SYN.SYN_REPORT) and dirty:
+                        dirty = False
+                        # len(styli) lines up, 1000 to the left and clear line
+                        print(f"\x1B[{len(styli)}A\x1B[1000D\x1B[K", end="")
+                        styli[(current_type, current_serial)] = in_prox
+                        for s, prox in styli.items():
+                            tid, serial = s
+                            print(
+                                f"Tool id {tid:#x} serial {serial:#x} in-proximity: {prox} "
+                            )
+        except KeyboardInterrupt:
+            print("Terminating")
+
+        return [s[0] for s in styli.keys()]
+
+
+def load_data_files():
+    stylusfile = Path('./data/libwacom.stylus')
+    if not stylusfile.exists():
+        stylusfile = Path('/usr/share/libwacom/libwacom.stylus')
+    if not stylusfile.exists():
+        die("Unable to find libwacom.stylus")
+    print(f'Using stylus file {stylusfile}')
```

Comment:
Should probably also add `/etc/libwacom` to the search path now that the library looks there.

Reply:
done. I've added the correct behaviour for the `prefix` now too, though there's a side-effect: if the prefix doesn't have files, we load from `/usr/share` anyway. Too niche to worry about, I reckon.